luxonis_train/attached_modules/metrics/object_keypoint_similarity.py

```diff
@@ -156,48 +169,115 @@ def compute(self) -> Tensor:
-            gt[:, None, :, 1] - pred[..., 1]
-        ) ** 2
-        kpt_mask = gt[..., 2] != 0  # only compute on visible keypoints
-        if self.use_cocoeval_oks:
-            # use same formula as in COCOEval script here:
-            # https://github.com/cocodataset/cocoapi/blob/8c9bcc3cf640524c4c20a9c40e89cb6a2f2fa0e9/PythonAPI/pycocotools/cocoeval.py#L229
-            oks = (
-                distances
-                / (2 * self.kpt_sigmas) ** 2
-                / (scales[:, None, None] + eps)
-                / 2
-            )
-        else:
-            # use same formula as defined here: https://cocodataset.org/#keypoints-eval
-            oks = (
-                distances
-                / ((scales[:, None, None] + eps) * self.kpt_sigmas.to(scales.device))
-                ** 2
-                / 2
-            )
 
-        return (torch.exp(-oks) * kpt_mask[:, None]).sum(-1) / (
-            kpt_mask.sum(-1)[:, None] + eps
+def compute_oks(
+    pred: Tensor,
+    gt: Tensor,
+    scales: Tensor,
+    sigmas: Tensor,
+    use_cocoeval_oks: bool,
+) -> Tensor:
+    """Compute Object Keypoint Similarity between every GT and prediction.
+
+    @type pred: Tensor[N, K, 3]
+    @param pred: Predicted keypoints.
+    @type gt: Tensor[M, K, 3]
+    @param gt: Groundtruth keypoints.
+    @type scales: Tensor[M]
+    @param scales: Scales of the bounding boxes.
+    @rtype: Tensor
+    @return: Object Keypoint Similarity every pred and gt [M, N]
+    @type sigmas: Tensor
+    @param sigmas: Sigma for each keypoint to weigh its importance, if C{None}, then use
+        same weights for all.
+    @type use_cocoeval_oks: bool
+    @param use_cocoeval_oks: Whether to use same OKS formula as in COCOeval or use the
+        one from definition.
+    """
+    eps = 1e-7
+    distances = (gt[:, None, :, 0] - pred[..., 0]) ** 2 + (
+        gt[:, None, :, 1] - pred[..., 1]
+    ) ** 2
+    kpt_mask = gt[..., 2] != 0  # only compute on visible keypoints
+    if use_cocoeval_oks:
+        # use same formula as in COCOEval script here:
+        # https://github.com/cocodataset/cocoapi/blob/8c9bcc3cf640524c4c20a9c40e89cb6a2f2fa0e9/PythonAPI/pycocotools/cocoeval.py#L229
+        oks = distances / (2 * sigmas) ** 2 / (scales[:, None, None] + eps) / 2
+    else:
+        # use same formula as defined here: https://cocodataset.org/#keypoints-eval
+        oks = (
+            distances
+            / ((scales[:, None, None] + eps) * sigmas.to(scales.device)) ** 2
+            / 2
         )
 
+    return (torch.exp(-oks) * kpt_mask[:, None]).sum(-1) / (
+        kpt_mask.sum(-1)[:, None] + eps
+    )
+
 
 def fix_empty_tensors(input_tensor: Tensor) -> Tensor:
     """Empty tensors can cause problems in DDP mode, this methods corrects them."""
     if input_tensor.numel() == 0 and input_tensor.ndim == 1:
         return input_tensor.unsqueeze(0)
     return input_tensor
+
+
+def set_sigmas(
```

Comment:
I'd rename it to `compute_sigmas` or `get_sigmas`. `set_sigmas` sounds like it should not return anything.